src/repart/repart.c

```diff
@@ -2458,6 +2468,88 @@ static int config_parse_encrypted_volume(
         return 0;
 }
 
+static int config_parse_tpm2_pcrs(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Partition *partition = ASSERT_PTR(data);
+        int r;
+
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                /* Clear existing PCR values if empty */
+                partition->tpm2_hash_pcr_values = mfree(partition->tpm2_hash_pcr_values);
+                partition->tpm2_n_hash_pcr_values = 0;
+                return 0;
+        }
+
+        r = tpm2_parse_pcr_argument_append(rvalue, &partition->tpm2_hash_pcr_values, &partition->tpm2_n_hash_pcr_values);
+        if (r < 0) {
+                return 0;
+        }
+
+        return 0;
+}
+
+static int parse_key_file(const char *filename, void **key, size_t *key_size) {
+        _cleanup_(erase_and_freep) char *k = NULL;
+        size_t n = 0;
+        int r;
+
+        r = read_full_file_full(
+                        AT_FDCWD, filename, UINT64_MAX, SIZE_MAX,
+                        READ_FULL_FILE_SECURE|READ_FULL_FILE_WARN_WORLD_READABLE|READ_FULL_FILE_CONNECT_SOCKET,
+                        NULL,
+                        &k, &n);
+        if (r < 0)
+                return log_error_errno(r, "Failed to read key file '%s': %m", filename);
+
+        erase_and_free(*key);
+        *key = TAKE_PTR(k);
+        *key_size = n;
+
+        return 0;
+}
+
+static int config_parse_key_file(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Partition *partition = ASSERT_PTR(userdata);
+        int r;
+
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                partition->key = erase_and_free(partition->key);
+                partition->key_size = 0;
+                return 0;
+        }
+
+        r = parse_key_file(rvalue, &partition->key, &partition->key_size);
```

Comment:
hmm, so when i think when reading the path from a repart.d/ drop-in then we should either strictly insist on absolute paths being specified.

we could eventually allow relative paths, but if so, we need to define clear what they are relative to (probably somethings like the --definitions= dir), and definitely not the cwd. i.e. i am pretty sure config files should never reference objects in a way that is subject to interpretation based on something so immediately part of the exec env.

Reply:
good point. I guess it's worth enforcing absolute path